pkg/webui/components/dropdown-v2/index.js

```diff
@@ -36,20 +36,24 @@ const Dropdown = ({ className, children, larger, onItemsClick }) => {
         const parentRect = ref.current.parentElement.getBoundingClientRect()
         const spaceBelow = window.innerHeight - parentRect.bottom
         const spaceAbove = parentRect.top
-        const spaceLeft = parentRect.left
-        const spaceRight = window.innerWidth - parentRect.right
+        const dropdownHeight = ref.current.clientHeight
+        const dropdownWidth = ref.current.clientWidth
+        const spaceOnLeft = parentRect.left
 
-        setIsBelow(spaceBelow > ref.current.clientHeight || spaceAbove < ref.current.clientHeight)
+        setIsBelow(spaceBelow > dropdownHeight || spaceAbove < dropdownHeight)
 
-        setIsOnRight(spaceRight < spaceLeft)
+        setIsOnRight(spaceOnLeft > dropdownWidth)
       }
     }
 
-    positionDropdown()
+    setTimeout(() => positionDropdown(), 200)
```

Comment:
On the initial render of the component(since now we are using visibility hidden) the position of the dropdown is not calculated correctly because the button component is still not displayed in it's final form.(You can see on the pictures that the `dropdownWidth` and `spaceOnLeft` are wrong without the `timeout` and the dropdown is wrongly positioned). I think this implementation won't be a problem since this is only a split of a second and helps to keep consistency.

- Before
![image](https://github.com/TheThingsNetwork/lorawan-stack/assets/49171937/2b551882-b8b9-4de9-b9ba-a99a4aff9d24)
![image](https://github.com/TheThingsNetwork/lorawan-stack/assets/49171937/ef0b9fe3-4964-4f9a-872f-0dbdf5f09c38)

- After 
![image](https://github.com/TheThingsNetwork/lorawan-stack/assets/49171937/07a8beb1-2aa2-4375-81c8-2c10c773099b)
![image](https://github.com/TheThingsNetwork/lorawan-stack/assets/49171937/9995dd6d-07c7-43d9-955a-e145c2463362)

Reply:
You can just use `useLayoutEffect` for that, which fires after all DOM mutations have been made:
```diff
diff --git a/pkg/webui/components/dropdown-v2/dropdown.styl b/pkg/webui/components/dropdown-v2/dropdown.styl
index 852958306..99ffa1205 100644
--- a/pkg/webui/components/dropdown-v2/dropdown.styl
+++ b/pkg/webui/components/dropdown-v2/dropdown.styl
@@ -26,12 +26,10 @@ ul.dropdown
   margin: $cs.xxs 0 0
   opacity: 0
   visibility: hidden
-  transition: opacity 0.2s ease-in-out, visibility 0s linear 0.2s
 
   &.open
     opacity: 1;
     visibility: visible;
-    transition: opacity 0.2s ease-in-out, visibility 0s linear
 
   &.below
     bottom: auto
diff --git a/pkg/webui/components/dropdown-v2/index.js b/pkg/webui/components/dropdown-v2/index.js
index aad60339c..1bd6fef20 100644
--- a/pkg/webui/components/dropdown-v2/index.js
+++ b/pkg/webui/components/dropdown-v2/index.js
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import React, { useCallback, useEffect, useRef, useState } from 'react'
+import React, { useCallback, useLayoutEffect, useRef, useState } from 'react'
 import classnames from 'classnames'
 import { NavLink } from 'react-router-dom'
 
@@ -30,23 +30,22 @@ const Dropdown = ({ className, children, larger, onItemsClick, open }) => {
   const [isBelow, setIsBelow] = useState(false)
   const [isOnRight, setIsOnRight] = useState(false)
 
-  useEffect(() => {
-    const positionDropdown = () => {
-      if (ref.current) {
-        const parentRect = ref.current.parentElement.getBoundingClientRect()
-        const spaceBelow = window.innerHeight - parentRect.bottom
-        const spaceAbove = parentRect.top
-        const dropdownHeight = ref.current.clientHeight
-        const dropdownWidth = ref.current.clientWidth
-        const spaceOnLeft = parentRect.left
-
-        setIsBelow(spaceBelow > dropdownHeight || spaceAbove < dropdownHeight)
-
-        setIsOnRight(spaceOnLeft > dropdownWidth)
-      }
+  const positionDropdown = useCallback(() => {
+    if (ref.current) {
+      const parentRect = ref.current.parentElement.getBoundingClientRect()
+      const spaceBelow = window.innerHeight - parentRect.bottom
+      const spaceAbove = parentRect.top
+      const dropdownHeight = ref.current.clientHeight
+      const dropdownWidth = ref.current.clientWidth
+      const spaceOnLeft = parentRect.left
+
+      setIsBelow(spaceBelow > dropdownHeight || spaceAbove < dropdownHeight)
+      setIsOnRight(spaceOnLeft > dropdownWidth)
     }
+  }, [])
 
-    setTimeout(() => positionDropdown(), 200)
+  useLayoutEffect(() => {
+    positionDropdown()
 
     window.addEventListener('scroll', positionDropdown)
     window.addEventListener('resize', positionDropdown)
@@ -55,7 +54,8 @@ const Dropdown = ({ className, children, larger, onItemsClick, open }) => {
       window.removeEventListener('resize', positionDropdown)
       window.removeEventListener('scroll', positionDropdown)
     }
-  }, [])
+  }, [positionDropdown])
+
   return (
     <ul
       onClick={onItemsClick}
```

Note that arbitrary timeouts are usually a code smell. This is because:
1. There's no guarantee the desired state will have settled after the timeout
2. There's potentially an unnecessary wait time (even if it's very small)
3. There's a potential memory leak risk if the timeout isn't cleared before the component unmounts

Reply:
It's still having the same issue on initial render using `useLayoutEffect`

![image](https://github.com/TheThingsNetwork/lorawan-stack/assets/49171937/59c775bf-55a0-47ff-92c6-5838d75de26c)

Reply:
> It's still having the same issue on initial render using `useLayoutEffect`.

Can't reproduce, what browser are you using?

Reply:
> Can't reproduce, what browser are you using?
It's only an edge case when we are on a tablet screen around 800px wide.

I came with a workaround and it works smoothly now. Please let me know what do you think about this approach.

Reply:
@kschiffer I tried refactoring this with a conditional rendering and without `visibility: hidden` but I couldn't manage to have the transition animation with that approach. So far the current solution seems to be working the best but let me know if you have some other suggestion which you think could make a better solution.

Reply:
It's ok, let's keep it as it is. I might go in and make a change later after we merge.